param_persist/agents/sqlalchemy_agent.py

```
@@ -0,0 +1,163 @@
+
+            for p in param_models:
+                db_session.add(p)
+
+            db_session.commit()
+
+        except Exception:
+            db_session.rollback()
+            raise
+        finally:
+            db_session.close()
+        return str(new_instance_uuid)
+
+    def load(self, instance_id):
+        """
+        Load an instance of a param object from the database.
+
+        Args:
+            instance_id: The id corresponding to the row in the database for the instance to load.
+
+        Returns:
+            The param instance populated from the database.
+        """
+        db_session = self.session_maker()
+
+        try:
+            instance_model = db_session.query(InstanceModel).filter_by(id=instance_id).first()
+            param_models = db_session.query(ParamModel).filter_by(instance_id=instance_id)
+            param_models_json = [x.value for x in param_models]
+            param_models_joined_json = ", ".join(param_models_json).replace("'", '"')
+            parameterized_json = '{' \
+                                 f'  "class_path": "{instance_model.class_path}",' \
+                                 f'  "params": [{param_models_joined_json}]' \
+                                 '}'
+            new_instance = ParamSerializer.from_json(parameterized_json)
+
+        except Exception:
+            db_session.rollback()
+            raise
+        finally:
+            db_session.close()
+
+        return new_instance
+
+    def delete(self, instance_id):
+        """
+        Delete an instance and its params from the database.
+
+        Args:
+            instance_id: The id of the instance to delete.
+        """
+        db_session = self.session_maker()
+
+        try:
+            instance_model = db_session.query(InstanceModel).get(instance_id)
+            if instance_model is None:
+                raise RuntimeError(f'unable to query database with given instance id. id="{instance_id}"')
```

Should we raise an exception on delete operations if the thing doesn't exist? I'd lean toward logging a warning and exiting early instead.

---

added logger